### PR TITLE
deluge 2 upgrade to python 3.10 fix for DSM 7

### LINKED ISSUE #4701: Deluge 2.0.3 not compatible with 3rd party plugins because it uses python 2.7 instead of 3.7
### Setup
_Package Name: deluge_ 
_Package Version: 2.0.3-13_ 

_NAS Model: DS120j_ 
_DSM version: DSM 6.2.4-25556_ 

### Expected behavior
Deluge uses python 3.7, which allows third party extension (build for deluge 2.x) to be installed.

### Actual behavior
Deluge 2.0.3 uses python uses Python 2.7. Third party extensions for deluge 2.x (packaged in egg format and build on python 3.x) do not work. The egg packages cannot be installed via the thin client, because the client compares the python version of deluge with the python version of the extension. The thin client version check can be circumvented by changing the version in the filename of the extension egg file from 3.x to 2.7. This allows the extension to be installed, but it won't work. python version differences.

### Steps to reproduce
_1. Download extension from https://dev.deluge-torrent.org/wiki/Plugins_ 
_2. In thin client: Preferences -> Plugins click Install plugin, locate the downloaded plugin egg and select it to install._

### LINKED ISSUE #5177: Deluge fails to install on DSM 7.0 on Armada 370
### Is this a new Bug?

- [X] I checkd that the bug hasn't been reported before

### Package Name

Deluge

### Package Version

2.0.3-14

### Device Model

DS115j

### Device Architecture

ARMv7

### Firmware Version

DSM 7.0.1-42218 Update 3

### What happened?

Deluge compiled from spksrc fails to install.
My best guess is that variables in service-setup are set improperly.

### Reproduction steps

```bash
1. Compile Deluge (currently 2.0.3-14) from spksrc.
2. Install Synology-published Python2 (currently 2.7.18-1004).
3. Deluge installation fails to detect Synology-published Python2.
4. Compile Python2 (currently 2.7.18-26) from spksrc.
5. Uninstall Synology-published Python2 and install spksrc-compiled Python2.
6. Deluge installation detects Python2 and attempts to install but fails.
```

### Install Log

```log
2022/03/26 13:30:34	install deluge 2.0.3-14 Begin preinst
2022/03/26 13:30:34	Begin reload_inst_variables
2022/03/26 13:30:34	End reload_inst_variables
2022/03/26 13:30:34	Begin initialize_variables
2022/03/26 13:30:34	End initialize_variables
2022/03/26 13:30:34	===> Step preinst. USER=sc-deluge GROUP=sc-download SHARE_PATH=/volume1/Downloads
2022/03/26 13:30:34	Begin service_preinst
2022/03/26 13:30:34	End service_preinst
2022/03/26 13:30:34	install deluge 2.0.3-14 End preinst ret=[0]
2022/03/26 13:30:34	install deluge 2.0.3-14 Begin /bin/mkdir -p /volume1/@appstore/deluge
2022/03/26 13:30:34	install deluge 2.0.3-14 End /bin/mkdir -p /volume1/@appstore/deluge ret=[0]
2022/03/26 13:30:34	install deluge 2.0.3-14 Begin /bin/rm -rf /volume1/@appstore/deluge
2022/03/26 13:30:34	install deluge 2.0.3-14 End /bin/rm -rf /volume1/@appstore/deluge ret=[0]
2022/03/26 13:30:34	install deluge 2.0.3-14 Begin /bin/mv -f /volume1/@tmp/pkginstall/extract.LnXhpx/package /volume1/@appstore/deluge
2022/03/26 13:30:34	install deluge 2.0.3-14 End /bin/mv -f /volume1/@tmp/pkginstall/extract.LnXhpx/package /volume1/@appstore/deluge ret=[0]
2022/03/26 13:30:34	install deluge 2.0.3-14 Begin /bin/rm -rf /var/packages/deluge
2022/03/26 13:30:34	install deluge 2.0.3-14 End /bin/rm -rf /var/packages/deluge ret=[0]
2022/03/26 13:30:34	install deluge 2.0.3-14 Begin /bin/mkdir -p /var/packages/deluge
2022/03/26 13:30:34	install deluge 2.0.3-14 End /bin/mkdir -p /var/packages/deluge ret=[0]
2022/03/26 13:30:34	install deluge 2.0.3-14 Begin /bin/touch /var/packages/deluge/installing
2022/03/26 13:30:34	install deluge 2.0.3-14 End /bin/touch /var/packages/deluge/installing ret=[0]
2022/03/26 13:30:34	install deluge 2.0.3-14 Begin /bin/mv -f /volume1/@tmp/pkginstall/extract.LnXhpx/INFO /var/packages/deluge/INFO
2022/03/26 13:30:35	install deluge 2.0.3-14 End /bin/mv -f /volume1/@tmp/pkginstall/extract.LnXhpx/INFO /var/packages/deluge/INFO ret=[0]
2022/03/26 13:30:35	install deluge 2.0.3-14 Begin /bin/rm -rf /var/packages/deluge/scripts
2022/03/26 13:30:35	install deluge 2.0.3-14 End /bin/rm -rf /var/packages/deluge/scripts ret=[0]
2022/03/26 13:30:35	install deluge 2.0.3-14 Begin /bin/mv -f /volume1/@tmp/pkginstall/extract.LnXhpx/scripts /var/packages/deluge/scripts
2022/03/26 13:30:35	install deluge 2.0.3-14 End /bin/mv -f /volume1/@tmp/pkginstall/extract.LnXhpx/scripts /var/packages/deluge/scripts ret=[0]
2022/03/26 13:30:35	install deluge 2.0.3-14 Begin /bin/rm -rf /var/packages/deluge/WIZARD_UIFILES
2022/03/26 13:30:35	install deluge 2.0.3-14 End /bin/rm -rf /var/packages/deluge/WIZARD_UIFILES ret=[0]
2022/03/26 13:30:35	install deluge 2.0.3-14 Begin /bin/mv -f /volume1/@tmp/pkginstall/extract.LnXhpx/WIZARD_UIFILES /var/packages/deluge/WIZARD_UIFILES
2022/03/26 13:30:35	install deluge 2.0.3-14 End /bin/mv -f /volume1/@tmp/pkginstall/extract.LnXhpx/WIZARD_UIFILES /var/packages/deluge/WIZARD_UIFILES ret=[0]
2022/03/26 13:30:35	install deluge 2.0.3-14 Begin /bin/rm -rf /var/packages/deluge/conf
2022/03/26 13:30:35	install deluge 2.0.3-14 End /bin/rm -rf /var/packages/deluge/conf ret=[0]
2022/03/26 13:30:35	install deluge 2.0.3-14 Begin /bin/mv -f /volume1/@tmp/pkginstall/extract.LnXhpx/conf /var/packages/deluge/conf
2022/03/26 13:30:35	install deluge 2.0.3-14 End /bin/mv -f /volume1/@tmp/pkginstall/extract.LnXhpx/conf /var/packages/deluge/conf ret=[0]
2022/03/26 13:30:41	install deluge 2.0.3-14 Begin postinst
2022/03/26 13:30:41	Begin reload_inst_variables
2022/03/26 13:30:41	End reload_inst_variables
2022/03/26 13:30:41	Begin initialize_variables
2022/03/26 13:30:41	End initialize_variables
2022/03/26 13:30:42	===> Step postinst. USER=sc-deluge GROUP=sc-download SHARE_PATH=/volume1/Downloads
2022/03/26 13:30:42	Begin save_wizard_variables
2022/03/26 13:30:42	End save_wizard_variables
2022/03/26 13:30:42	Begin syno_sync_var_folder
2022/03/26 13:30:42	Install files from var folder
2022/03/26 13:30:42	/bin/rsync -avh --ignore-existing --remove-source-files /volume1/@appstore/deluge/var/ /volume1/@appdata/deluge
2022/03/26 13:30:42	sending incremental file list
2022/03/26 13:30:42	./
2022/03/26 13:30:42	autoadd.conf
2022/03/26 13:30:42	core.conf
2022/03/26 13:30:42	sent 4.19K bytes  received 73 bytes  8.53K bytes/sec
2022/03/26 13:30:42	total size is 4.00K  speedup is 0.94
2022/03/26 13:30:42	/bin/rsync -avh --remove-source-files /volume1/@appstore/deluge/var/ /volume1/@appdata/deluge
2022/03/26 13:30:42	sending incremental file list
2022/03/26 13:30:42	./
2022/03/26 13:30:42	sent 71 bytes  received 19 bytes  60.00 bytes/sec
2022/03/26 13:30:42	total size is 0  speedup is 0.00
2022/03/26 13:30:42	End syno_sync_var_folder
2022/03/26 13:30:42	Begin service_postinst
2022/03/26 13:30:42	/var/packages/deluge/scripts/service-setup: line 59: /var/log/packages/deluge.log: Permission denied
2022/03/26 13:30:42	/var/packages/deluge/scripts/service-setup: line 62: /var/log/packages/deluge.log: Permission denied
2022/03/26 13:30:42	/var/packages/deluge/scripts/service-setup: line 65: /var/log/packages/deluge.log: Permission denied
2022/03/26 13:30:42	Notice: set_unix_permissions() is no longer required on DSM7.
2022/03/26 13:30:42	sed: can't read /volume1/@appstore/deluge/var/core.conf: No such file or directory
2022/03/26 13:30:42	sed: can't read /volume1/@appstore/deluge/var/autoadd.conf: No such file or directory
2022/03/26 13:30:42	sed: can't read /volume1/@appstore/deluge/var/core.conf: No such file or directory
2022/03/26 13:30:42	sed: can't read /volume1/@appstore/deluge/var/autoadd.conf: No such file or directory
2022/03/26 13:30:42	sed: can't read /volume1/@appstore/deluge/var/core.conf: No such file or directory
2022/03/26 13:30:42	sed: can't read /volume1/@appstore/deluge/var/autoadd.conf: No such file or directory
2022/03/26 13:30:42	sed: can't read /volume1/@appstore/deluge/var/autoadd.conf: No such file or directory
2022/03/26 13:30:42	/var/packages/deluge/scripts/service-setup: line 92: /var/log/packages/deluge.log: Permission denied
2022/03/26 13:30:42	/var/packages/deluge/scripts/service-setup: line 93: /var/log/packages/deluge.log: Permission denied
2022/03/26 13:30:42	End service_postinst
2022/03/26 13:30:42	install deluge 2.0.3-14 End postinst ret=[0]
```

### Service Log

_No response_

### Other Logs

_No response_

### PR DESCRIPTION
**Pull request by Strangelovian**
_Motivation:_  Have the deluge 2.0 upgrade functional with python 3.7
_Linked issues:_  #4134, #4158, closes #4701, hopefully closes #5177

Update as deluge 2.0.5 was officially released in Dec 2021
- [ ] Update Deluge to v 2.0.5
- [ ] Update to Python 3.10
- [ ] Update for DSM 7
- [ ] No need for prerelease by @Strangelovian (tanks for investigation anyway).
- [ ] Finally libtorrent was updated to v1.12.10 (further updates, or updates to libtorrent v2 are out of scope here, and probably not compatible with the synology toolchains).

Notes:
- Cross compilation works for at least: x64, aarch64, denverton
- x64 and denverton packages are confirmed working on DS1819+: install OK, download torrent OK, uninstall OK
- the core.conf was changed to have initially a single port configured, else deluge 2 will not managed to bind any port. Deluge 2 manages only one listen port. The core.conf still accepts a list of listen ports in the core.conf for compatibility reasons with deluge 1.x.
- I updated the package icon to new deluge 2 icon
- boost upgrade is needed from v1.63 to v1.67, else Python3.Boost fails to compile
- cross compilation fails with boost v1.74 for aarch64. Fortunately v1.67 works with all architectures

**CAVEAT PLEASE READ:**
deluge daemon v2.0.3 won't start because of this issue: https://dev.deluge-torrent.org/ticket/3278
The fix is already done here https://github.com/deluge-torrent/deluge/commit/1b4ac88ce72515bcf36684b942e3998900f13cea
But no release was made since 2.0.3 so I forked it to have the last update: https://github.com/Strangelovian/deluge/tree/2.0.4-str0
Now, I will ask the deluge team to build an official 2.0.4  to avoid using my fork. https://forum.deluge-torrent.org/viewtopic.php?f=8&t=55767
Normally this should be quick to do, even though the deluge developments have really slowed down recently.

### Checklist
- [X] Build rule `all-supported` completed successfully
- [X] Package upgrade completed successfully
- [X] New installation of package completed successfully

upgrade to boost 1.67

upgrade to libtorrent 1.2.9

build from deluge bleeding edge 2.0.4

update deluge icon

adjust core.config with a single port, as required by deluge version 2.x

### CODE REVIEW

**Comment by Strangelovian** [comment]
Regarding rutorrent failing build for all archs (!):
- I'm unsure why it was selected to be tested by the continuous integration jobs, it shouldn't: only boost version, libtorrent-rastebar (not libtorrent-rakshasa!) have been changed versions in the cross folder. Rutorrent don't depend on that...
- Here the compile error with rutorrent:
```
wget https://github.com/rakshasa/libtorrent/archive/v0.13.6.tar.gz
2020-08-30 21:54:04 URL:https://codeload.github.com/rakshasa/libtorrent/tar.gz/v0.13.6 [340818] -> "libtorrent-0.13.6.tar.gz.part" [1]
...
===>  Extracting for libtorrent
tar -xzpf /spksrc/cross/libtorrent-rakshasa/../../distrib/libtorrent-0.13.6.tar.gz -C /spksrc/spk/rutorrent/work-x64-6.1
===>  Patching for libtorrent
aclocal...
configure.ac:22: warning: macro 'AM_PATH_CPPUNIT' not found in library
autoheader...
libtoolize... using libtoolize
automake...
configure.ac:3: installing './compile'
configure.ac:3: installing './config.guess'
configure.ac:3: installing './config.sub'
configure.ac:20: installing './install-sh'
configure.ac:20: installing './missing'
src/Makefile.am: installing './depcomp'
parallel-tests: installing './test-driver'
autoconf...
configure.ac:22: error: possibly undefined macro: AM_PATH_CPPUNIT
      If this token and others are legitimate, please use m4_pattern_allow.
      See the Autoconf documentation.
```
This is quite weird, no dependencies of cross/libtorrent-rakshasa have been changed by this commit.

**Comment by Strangelovian** [comment]
Regarding libtorrent 1.2.9 build failing for arch qoriq:
```
  CXX      libtorrent_rasterbar_la-enum_net.lo
enum_net.cpp: In function 'libtorrent::{anonymous}::link_info libtorrent::{anonymous}::parse_nl_link(const nlmsghdr*)':
enum_net.cpp:341:10: error: 'IFLA_CARRIER' was not declared in this scope
     case IFLA_CARRIER:
          ^
enum_net.cpp:352:10: error: 'IFLA_STATS64' was not declared in this scope
     case IFLA_STATS64:
          ^
enum_net.cpp:354:10: error: 'IFLA_PROMISCUITY' was not declared in this scope
     case IFLA_PROMISCUITY:
          ^
```
It seems that some linux network enumerations are not up to date in the qoriq toolchain (https://docs.huihoo.com/doxygen/linux/kernel/3.7/uapi_2linux_2if__link_8h.html)

**Comment by Strangelovian** [comment]
For arch evansport, nearly same issue as for arch qoriq during libtorrent 1.2.9 build:

```
  CXX      libtorrent_rasterbar_la-enum_net.lo
enum_net.cpp: In function ‘libtorrent::{anonymous}::link_info libtorrent::{anonymous}::parse_nl_link(const nlmsghdr*)’:
enum_net.cpp:341:10: error: ‘IFLA_CARRIER’ was not declared in this scope
     case IFLA_CARRIER:
          ^
enum_net.cpp:354:10: error: ‘IFLA_PROMISCUITY’ was not declared in this scope
     case IFLA_PROMISCUITY:
          ^
```

**Comment by Strangelovian** [comment]
FTR, the rutorrent builds are already broken as of current master branch, currently at:
```
commit 4593f8173219a776c89621ef64aeeaefc48b42aa (HEAD -> master, origin/master
, origin/HEAD)
Author: Vincent Fortier <th0ma7@users.noreply.github.com>
Date:   Thu Aug 20 07:13:56 2020 -0400
    x265: Use github mirror as mercurial url is now dead (#4140)
```

**Comment by Strangelovian** [comment]
link issue with rutorrent compilation https://github.com/SynoCommunity/spksrc/issues/4158

**Comment by ymartin59** [comment]
@hgy59 I request your opinion about Boost upgrade. If I remember well, version is stuck because of others applications, right?

**Comment by ymartin59** [comment]
Deluge wizard provides instruction I wonder if it is still relevant for version 2: `To allow clients to remotely connect to the daemon, see <a target=\"blank\" href=\"https://dev.deluge-torrent.org/wiki/UserGuide/ThinClient\">the ThinClient guide of Deluge</a>`
May you please check to confirm if this should be discarded or not from wizard.
By the way, I would prefer to have details in a dedicated wiki page at https://github.com/SynoCommunity/spksrc/wiki/Package-Documentation-Index

**Comment by Strangelovian** [comment]
> Deluge wizard provides instruction I wonder if it is still relevant for version 2: `To allow clients to remotely connect to the daemon, see <a target=\"blank\" href=\"https://dev.deluge-torrent.org/wiki/UserGuide/ThinClient\">the ThinClient guide of Deluge</a>`
> May you please check to confirm if this should be discarded or not from wizard.
> By the way, I would prefer to have details in a dedicated wiki page at https://github.com/SynoCommunity/spksrc/wiki/Package-Documentation-Index

Deluge v2.x  is still based on daemon + client. Of course, daemon + delugeweb client still works. Most people will use this with synology nases, as the syno spksrc packages both.
I didn't try to start daemon + gtk thin UI. First of all, the gtk UI is no shipped on windows since deluge v2, for a problem a gtk library...
If you ask me, I would rather remove the thin client link.

**Comment by Strangelovian** [comment]
check fails details as of 10/25/2020:
- qoriq-6.1 / evansport-6.1: missing linux header if_link.h IFLA_CARRIER IFLA_PROMISCUITY in toolkit
- ppc853x-5.2 / 88f6281-6.1: saltminion python 3.7 build fails (Building wheel for pyzmq)

@ymartin59 @hgy59 may I have your opinion about this?

**Comment by hgy59** [comment]
>     * ppc853x-5.2 / 88f6281-6.1: saltminion python 3.7 build fails (Building wheel for pyzmq)

Don't worry about `salt-minion`. These archs failed since the package was added with #4119. I don't know why this PR was merged without defining older ppc and ARM5 archs as unsupported.

**Comment by ymartin59** [comment]
@hgy59 My fault.... we should create a common "review checklist" in wiki so that no point is missed.

**Comment by Strangelovian** [comment]
> 
> 
> @hgy59 My fault.... we should create a common "review checklist" in wiki so that no point is missed.

@ymartin59  @hgy59 ok about that salt-minion. About qoriq and evansport, is it OK to drop support because of libtorrent? Maybe it's going to improve with DSM7 for those 2 archs, if they get a more recent cross compilation toolkit. What do you think?

**Comment by Strangelovian** [comment]
@ymartin59 @hgy59 after clean-up of unsupported arches, clean-up of wizard text, all seems pretty much OK. Ready for review?

**Comment by zyrill** [comment]
@hgy59 anything that still needs doing here that I can help with?

**Comment by hgy59** [comment]
> @hgy59 anything that still needs doing here that I can help with?

There is no official v2.0.4 yet. 
I propose to leave this PR untouched for experimantal builds by interested users, until a new release is created in the official repo (if ever ?).

_for me the deluge projects seems dead, as since 2019-06-12 there is no aticivity under https://git.deluge-torrent.org/deluge_

**Comment by zyrill** [comment]
@hgy59 you're right about the project being abandoned - so wouldn't it then make sense to also close this ticket? Interested parties could still search for it but maybe nobody should be encouraged to actually use it. Plus this would clean up one more item on the open PR list.

File: cross/deluge/digests
Comment (ymartin59) [inline cross/deluge/digests:3]: Required. Please enforce file name as "deluge-2..." with PKG_DIST_FILE variable

File: spk/deluge/Makefile
Comment (ymartin59) [inline spk/deluge/Makefile:24]: Is date still relevant?

File: spk/deluge/Makefile
Comment (ymartin59) [inline spk/deluge/Makefile:3]: Required value > 13, so set it to 14

File: spk/deluge/src/core.conf
Comment (ymartin59) [inline spk/deluge/src/core.conf:52]: No need for two lines with same port number. What do you expect here?

**Review by ymartin59 — APPROVED** [review]
Thanks for contribution. I have minor improvements proposals.

**Comment by hgy59** [comment]
as #4539 is now on the master branch, this PR needs some rework.
all  ` >> ${INST_LOG}`  in `service-setup.sh` have to be removed for service_postinst to work.
and `service_preinst` must be renamed to `validate_preinst` to get the same output for `echo` in this function.

File: spk/deluge/src/core.conf
Comment (hgy59) [reply spk/deluge/src/core.conf:52]: `listen_ports` defines a range of ports [ from_port, to_port ] so it needs two entries for a single port too.

File: cross/libtorrent/Makefile
Comment (th0ma7) [inline cross/libtorrent/Makefile:10]: Could this be dependent to python310?  Also OpenSSL is already a dependency of python so a repeat of it is probably unecessary (while being informative)

Comment (hgy59) [reply cross/libtorrent/Makefile:10]: Currently I have troubles with `cross/libtorrent` that is a _double target_ build. One builds the libtorrent and the other target builds the python-libtorrent wheel. And the prefered build tool is boost b2, ...

Yes I tried python310 but currently trying python38, as I got errors like:
```
configure: error: 
  Could not link test program to Python. Maybe the main Python library has been...
```
This error may be due to the fact, that it was never possible to build wheels in the cross folder.

The main challenge is to cross compile the wheel.  As there is no wheel source available neither for deluge-libtorrent nor python-libtorrent.

As I have fixed the python-wheel target name (was renamed from `build_python_wheel`to `build_python_wheel_target`)
The build now stops with:

```
/bin/sh: 1: Syntax error: "(" unexpected
make[3]: *** [../../mk/spksrc.python-wheel.mk:39: build_python_wheel_target] Error 2
```

but a second run of make in spk/deluge folder succeeds...

So I will try to use python310 first, and then fix the error above...

**Comment by hgy59** [comment]
@th0ma7 Cannot use Python 3.10 for this package, as boost (1.66.0) python bindings fail to build with v3.10, but work with v3.8:
```
gcc.compile.c++ bin.v2/libs/python/build/gcc-gnu-4.9.4/release/link-static/threadapi-pthread/threading-multi/exec.o
libs/python/src/exec.cpp: In function ‘boost::python::api::object boost::python::exec_file(const char*, boost::python::api::object, boost::python::api::object)’:
libs/python/src/exec.cpp:109:30: error: ‘_Py_fopen’ was not declared in this scope
   FILE *fs = _Py_fopen(f, "r");
                              ^
```

File: cross/libtorrent/Makefile
Comment (hgy59) [reply cross/libtorrent/Makefile]: > Could this be dependent to python310? Also OpenSSL is already a dependency of python so a repeat of it is probably unecessary (while being informative)

Finally it builds with python310. 
python is BUILD_DEPENDS only but openssl is a real dependency. - so it must not be removed.

BTW repeated dependencies are no issue, as those are skipped anyway (recognized by the cookie files).

And even when dependencies are already included by dependent packages, it is IMHO not recommended to remove _unnecessary_ dependencies for the sake of information of what is required by a package (and to avoid unexpected errors when dependencies of dependent packages change...)

**Comment by th0ma7** [comment]
@hgy59 I had completely forgoten but I still have pending patches for this PR but I wasn't able to push them in / access denied I'll try to revisit this for a bit and figure it out and push them in over the next week-end or so.

On another note, are you able to rebase against master so the code is using the latest changes?

**Comment by hgy59** [comment]
> On another note, are you able to rebase against master so the code is using the latest changes?

It was already rebased to b07fbcc802085829679f4d0890edfc524a373dfb two days a go.
now I have rebased to f79e12f0e8eb28de9a07a9d3699c2498db01135d.

**Comment by hgy59** [comment]
> but I still have pending patches for this PR

@th0ma7 I could not find any branch in your repo related to delugepy. 
Can you tell me on which of your banches / which commit are for delugepy? I will try to cherry pick those into this PR.

**Comment by Safihre** [comment]
@hgy59 You seem to be getting the same problem as me with SABnzbd, cryptography suddenly fails to build..

**Comment by th0ma7** [comment]
@hgy59 seems I was able to push changes in... Seems I had first to use `gh` to get things up.

**Comment by th0ma7** [comment]
@Safihre using your changes from #5316 fixed building on my local tree :)
@hgy59 feel free to rebase once @Safihre PR is merged!

**Comment by th0ma7** [comment]
@hgy59 what's now left is getting rid of `dsm-control.sh`.  At first glance two options comes to mind:
1. Modify `start-stop-status` to allow managing multiple daemon startups.  This could be done using a default delimiter such as `:` or using `\n` and parsing each lines as separate daemons.  This also leads to adapting the `PID_FILE` creation in order to concatenate pid's.
2. Use something similar to `rutorrent`: default `start-stop-status` script refers to usual `SERVICE_COMMAND` but it calls another script to do the work.  That other script could be of a similar nature and would start-up both daemons and adjust pid file as well.

I think option 1) would bring more benefit as it can be reused on any package aftewards.  On the other hand we must be careful as depending how it's done it may impact spechial caracters (e.g. `\something`) although I haven't noticed any of that.  On the other hand option 2) could be easier/faster to implement.

I'll give it a shot at option 1) and see how it turns out.  Worst case we can simply undo it afterwards.

**Comment by th0ma7** [comment]
A follow-up, the good:
- Option 1. looks relatively simple to implement.  Got some code I'm currently playing with pending further testing/debug.

The bad:
- `libboost_python310.so.1.70.0` is not being packaged... actually not even being copied over to the `target/lib` at the end of boost build/installation.  So there is that to fix either by further changing boost 1.70 or switching to a newer boost version.
- After manually copying it over and restarting deluge, it currently segfault core-dump on `deluged`.  Perhaps due to incompatible boost library?

**Comment by th0ma7** [comment]
A subsequent follow-up:

**The good:**
- got a deluge 2.0.5 + python310 + boost 1.78 + libtorrent 2.0.6 working (e.g. loads to memory properly)
This is only good for DSM7 as c++14 is needed (and fails with c++1y equivalent on gcc-4.9)
Code is on my local drive, nothing pushed up yet as I have more testing to do
I still have to test building boost 1.70 while enforcing `c++1y` using gcc-4.9 then try to build libtorrent 2.x ... 
- got a deluge 2.0.5 + python310 + boost 1.70 + libtorrent 1.2.16 building OK (but segfault, at least on DSM6)

**The bad:**
- Latest version of `boost` officially supporting  c++11 from gcc<5.1 is boost 1.70 (1.71 might work, haven't tried but fails starting with 1.72).  This mean DSM6 = boost 1.70ish
- boost 1.70 doesn't support python310.  While we do have a patch there is more needed to get it working (e.g. getting `libboost_python310.so.1.70.0` installed for a start).

**The ugly :**
Our options are:
1. only use libtorrent 1.2 + boost 1.70 - but segfault needs fixing
2. use different versions for DSM6 vs DSM7
3. only support DSM7

**Comment by FloBer33** [comment]
@th0ma7 as a DSM7 user, Deluge is really missing me since the upgrade, wouldn't it be better to have a version supporting the latest version of DSM only rather than a version less and less used and outdated ?

**Comment by th0ma7** [comment]
@Strangelovian can you please allow edit from maintainers to simplify the process of updating this PR?  Thnx in advance.

**Comment by Strangelovian** [comment]
> @Strangelovian can you please allow edit from maintainers to simplify the process of updating this PR? Thnx in advance.

it's already allowed, please check

**Comment by th0ma7** [comment]
Thnx but I was having too many issues trying to update from your PR.  As such I've started a nwe one and imported all changes as co-authored commit and resumed work from there  using a clean and rebased branch (PR #5398).  It is now much closer from a release as latest version of binaries do compile and run properly (while still missing proper startup mechanism).

Feel free to chim-in if you have spare cycles and thnx again for your efforts on this.

**Pull request closed without merging**
